pkg/controller/volume/persistentvolume/metrics/metrics.go

```diff
@@ -122,33 +147,55 @@ func (collector *pvAndPVCCountCollector) CollectWithStability(ch chan<- metrics.
 	collector.pvcCollect(ch)
 }
 
+func (collector *pvAndPVCCountCollector) getPVPluginName(pv *v1.PersistentVolume) string {
+	spec := volume.NewSpecFromPersistentVolume(pv, true)
+	fullPluginName := pluginNameNotAvailable
+	if plugin, err := collector.pluginMgr.FindPluginBySpec(spec); err == nil {
+		fullPluginName = metricutil.GetFullQualifiedPluginNameForVolume(plugin.GetPluginName(), spec)
+	}
+	return fullPluginName
+}
+
 func (collector *pvAndPVCCountCollector) pvCollect(ch chan<- metrics.Metric) {
-	boundNumberByStorageClass := make(map[string]int)
-	unboundNumberByStorageClass := make(map[string]int)
+	boundNumberByPlugin := make(volumeCount)
+	unboundNumberByPlugin := make(volumeCount)
 	for _, pvObj := range collector.pvLister.List() {
 		pv, ok := pvObj.(*v1.PersistentVolume)
 		if !ok {
 			continue
 		}
+		pluginName := collector.getPVPluginName(pv)
 		if pv.Status.Phase == v1.VolumeBound {
-			boundNumberByStorageClass[pv.Spec.StorageClassName]++
+			boundNumberByPlugin.add(pluginName, pv.Spec.StorageClassName, string(*pv.Spec.VolumeMode))
 		} else {
-			unboundNumberByStorageClass[pv.Spec.StorageClassName]++
+			unboundNumberByPlugin.add(pluginName, pv.Spec.StorageClassName, string(*pv.Spec.VolumeMode))
 		}
 	}
-	for storageClassName, number := range boundNumberByStorageClass {
-		ch <- metrics.NewLazyConstMetric(
-			boundPVCountDesc,
-			metrics.GaugeValue,
-			float64(number),
-			storageClassName)
+	for pluginName, storageClassCount := range boundNumberByPlugin {
+		for storageClassName, volumeModeCount := range storageClassCount {
+			for volumeMode, number := range volumeModeCount {
+				ch <- metrics.NewLazyConstMetric(
+					boundPVCountDesc,
+					metrics.GaugeValue,
+					float64(number),
+					storageClassName,
+					pluginName,
+					volumeMode)
+			}
+		}
 	}
-	for storageClassName, number := range unboundNumberByStorageClass {
-		ch <- metrics.NewLazyConstMetric(
-			unboundPVCountDesc,
-			metrics.GaugeValue,
-			float64(number),
-			storageClassName)
+	for pluginName, storageClassCount := range unboundNumberByPlugin {
+		for storageClassName, volumeModeCount := range storageClassCount {
+			for volumeMode, number := range volumeModeCount {
+				ch <- metrics.NewLazyConstMetric(
+					unboundPVCountDesc,
+					metrics.GaugeValue,
+					float64(number),
+					storageClassName,
+					pluginName,
+					volumeMode)
+			}
+		}
 	}
```

Comment:
some new unit tests for checking these labels would be nice.

Reply:
You're right... I don't see any unit tests for the metrics but there's e2e that might be extended:
https://github.com/openshift/kubernetes/blob/master/test/e2e/storage/volume_metrics.go#L381-L509